File: RecipeBook.Api/Models/Category.cs
namespace RecipeBook.Api.Models
{
    public class Category
    {
        public string? Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Color { get; set; }
        public int DisplayOrder { get; set; }
        public int RecipeCount { get; set; }  // Computed field
    }

    public class CategoryCreateRequest
    {
        public string Name { get; set; } = null!;
        public string? Color { get; set; }
    }

    public class CategoryUpdateRequest
    {
        public string Name { get; set; } = null!;
        public string? Color { get; set; }
        public int DisplayOrder { get; set; }
    }
}

File: recipebook.web/src/models/category.ts
export interface Category {
    id: string;
    name: string;
    color?: string;
    displayOrder: number;
    recipeCount: number;
}

export interface CategoryCreateRequest {
    name: string;
    color?: string;
}

export interface CategoryUpdateRequest {
    name: string;
    color?: string;
    displayOrder: number;
}
